testdata/p4_16_samples_outputs/psa-example-counters-bmv2-frontend.p4

```diff
@@ -103,7 +103,7 @@ control ingress(inout headers hdr, inout metadata user_meta, in psa_ingress_inpu
             psa_ingress_output_metadata_t meta_2 = ostd;
             PortId_t egress_port_1 = oport;
             meta_2.drop = false;
-            meta_2.multicast_group = 10w0;
+            meta_2.multicast_group = (MulticastGroup_t)32w0;
```

Comment:
newtype was introduced to generate efficient control plane API. Since control plane software typically communicates with the dataplane through a PCIe bus, and the PCIe bus protocol work at the granularity of bytes. The common use case for newtype is to cast a non-byte aligned parameter to byte-aligned parameter for control plane API generation. However, this casting should not introduce overhead in dataplane. The non-byte aligned parameter should remain non-byte aligned in the dataplane, as that is the most efficient representation in the dataplane. The changes here looks problematic to me, as it changes the size of metadata from PortId_t to bit<32>.

Reply:
ah, Andy changed all the _uint_t to be byte-aligned. @jafingerhut there isn't much benefit to use `type` if everything is byte-aligned, no?

Reply:
I would recommend reading the PSA v1.1 spec, Sections 4.1 and 4.4, to see if it answers any of your questions for why `type` can be useful for PSA programs for defining types like `PortId_t`.  I am happy to answer questions about it after you have read those, in case they do not fully answer your questions, but I am hoping they might.

There was a strong desire by multiple people involved in PSA and P4Runtime to enable numeric runtime translation between the control plane and the data plane, as well as custom bit widths in the data plane.  `type` was added to P4_16 so that we could have distinct types for things like `PortId_t` that are preserved until the control plane API is generated, and the name `PortId_t` can be _included_ in the control plane API, i.e. in this case the P4Info file, wherever a value of that type occurs, so that the P4Runtime client and server know where in the API all such values appear.

Also these bit widths are BMv2-specific bit widths, as mentioned in comments here in the psa.p4 include file here: https://github.com/p4lang/p4c/blob/master/p4include/psa.p4#L34-L40

A different PSA implementation P1 can choose a 9-bit width PortId_t type in the data plane if they want, and yet another P2 could choose to use a 6-bit PortId_t type in the data plane.